src/libs/ajax/AzureStorage.ts

```diff
@@ -64,21 +79,29 @@ export const AzureStorage = signal => ({
     const xml = new window.DOMParser().parseFromString(text, 'text/xml')
     const blobs = _.map(
       blob => ({
-        name: _.head(blob.getElementsByTagName('Name')).textContent,
+        name: _.head(blob.getElementsByTagName('Name'))?.textContent || '',
         lastModified: new Date(
-          _.head(blob.getElementsByTagName('Last-Modified')).textContent
+          _.head(blob.getElementsByTagName('Last-Modified'))?.textContent || '0'
```

Comment:
I wonder if it'd be helpful to define types encoding `?.textContent || ''` and `?.textContent || '0'` such that they're contained in a more `Decoder`y place (like Scala decoders in Leo code base).

Reply:
> After an initial scan, a few themes:
> 
> * Functions in TypeScript modules should have type annotations for parameters. I'm fine with gradually typing the large Ajax modules, but the small, analysis related, file/tool utils modules could certainly be typed here.
> * Tests generally don't need to use `act`.
> * Tests contain a lot of unnecessary awaits.
> * In most cases, ts-expect-error is probably preferable to ts-ignore since we want to improve type validation to recognize those cases in the future.

still working on this part....